### ISSUE TITLE
Make the gallery the primary API reference doc for angular_components

### ISSUE THREAD

**Issue by kwalrath** [dart-lang/site-webdev#1533]
Related:

* https://github.com/dart-lang/angular_components/issues/217#issuecomment-384026372
* https://github.com/dart-lang/angular_components/issues/242

**Comment by chalin** [comment]
I'm not sure that I understand this request. The gallery doesn't contain any API reference entries. In fact, in the landing page **Resources >> API reference** points to https://webdev.dartlang.org/components/api.

**Comment by kwalrath** [comment]
The gallery contains HTML-oriented reference docs that are generated from the class descriptions. E.g. if you go to https://dart-lang.github.io/angular_components_example/#/app_layout, there's reference information from class-level doc comments.

**Comment by chalin** [comment]
Right, but suppose you come across a `<glyph>` in an ACX app's HTML: if you [type glyph in the search box](https://dart-lang.github.io/angular_components_example/), nothing is shown. What then should the reader do?

I believe that this issue has a few parts, the main one being:

- Update references to ACX API so that readers are referred to https://dart-lang.github.io/angular_components_example

I think that to have a fuller UX we would need to enhance the ACX example app so that, for example:

- Every page provides a link back to the full API.
- Maybe the search could cover the full API.

Finally:

- We need to decide where the full API docs will be hosted from.

**Comment by kwalrath** [comment]
@nshahan & @TedSander, do you have feedback on Patrice's comment?

**Comment by TedSander** [comment]
So I think those are all good points, we aren't saying that the gallery is currently perfect in terms of documentation, but we think it will be a better landing page for angular components than the API documentation.

There are a couple of reasons:
a) The Dart API is not the api that we want used for interaction of a component. Many things are made public for angular templates, but they aren't for general consumption. the @Input/@Outputs/@Attributes ng-content are.
b) The examples and demos are very useful for knowing what to do with components. Internally we don't even have DartDoc, but the users find the gallery invaluable. It is consistently one of things they say is best about the widgets.
c) Mixins/Readmes provide a lot of the documentation for the components and I don't see these going into the API doc, but they are on the roadmap for the gallery.

As for the search box I like the idea of adding the selector. Glyph does not show up today because it is deprecated, but class names are already searchable (MaterialIconComponent for instance.) Please file a feature request and assign it to Nick for this.

It would also be good to fill in the holes around documentation that isn't in the gallery. Which should be relatively easy to do.

tl;dr; The gallery is an excellent internal resource where engineers go for documentation. Between investing in making the API docs better or the gallery better I see the gallery giving the better experience in the long run for users.

**Comment by nshahan** [comment]
I like the idea of adding selectors as additional search terms and that sounds like a relatively small change to the current gallery generators.

There is another gap in documentation that I haven't been able to come up with a good solution for. There are plenty of classes in the angular_components package that are not` @Component`s and don't have a page in the gallery. These are things like the model classes, utilities, etc. Discovering those files and getting documentation is difficult.

Internally we have discussions groups where I see engineers go for help working with those. 

I could look into adding a link to the dartdoc API page for each `@Component` in the gallery but I have been reluctant to since we don't have a version of that site that I could point to internally. I suppose it could be an external only feature.

**Comment by TedSander** [comment]
Internal can always point externally. It might be a bit out of sync but not
too bad. Some classes might not have links if they we're internal only but
that is better than nothing.

**Comment by nshahan** [comment]
@kwalrath @chalin If we were to link to the API docs, is there a link we can use to always point to the most recently published version? The gallery is always built with the latest published version regardless of alpha, beta etc. The link https://pub.dartlang.org/documentation/angular_components/latest/ always points to the most recently published "stable" version which is from October 2017 at this point.

**Comment by chalin** [comment]
@TedSander - that all SGTM.

> I like the idea of adding the selector. ... Please file a feature request

Done: https://github.com/dart-lang/angular_components_example/issues/79.

>  ... is there a link we can use to always point to the most recently published version?

Hmm, I'm sure that I came across a way to do this (by putting `dev` somewhere in the path), but I can't get that to work atm. Anyhow, I think that it would be **preferable to link to the version of angular_components that the app was built from**.

Another idea would be to **publish `angular_components_example` via webdev.dartlang.org**. In this way we can keep versions associated with the appropriate sites: https://webdev.dartlang.org for the latest, and https://webdev-dartlang-org-dev.firebaseapp.com/ for dev releases.

WDYT?

**Comment by chalin** [comment]
PR #1559 is what it would take to deploy the ACX gallery under the webdev site. Let me know what you  think.

**Comment by nshahan** [comment]
@chalin What would the deploy process be like on every update? 

@kevmoo what would this mean for the analytics?

**Comment by kwalrath** [comment]
So... Are we good with hosting the dev & stable releases of the gallery on webdev.dartlang.org, so that it's versioned and we can easily link to it (and can create redirects, when necessary)?

**Comment by kevmoo** [comment]
Kathy – let's chat about this next week. And have Nick VC in...
